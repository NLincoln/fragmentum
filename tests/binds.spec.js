import { builder, select, from, where, bind, ops } from "fragmentum";

describe("variable binds", () => {
  test("can bind variables", () => {
    const sql = builder(
      select(),
      from("users"),
      where(ops.eq("id", bind("user_id", 3)))
    ).serialize();
    expect(sql).toHaveProperty("binds");
    expect(sql).toHaveProperty("query");
    expect(sql).toMatchSnapshot();
  });
  test("value() will return a bound var with an anonymous key");
});
