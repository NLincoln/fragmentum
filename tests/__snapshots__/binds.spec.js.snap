// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`variable binds can bind variables 1`] = `
Object {
  "binds": Array [
    Object {
      "user_id": 3,
    },
  ],
  "query": "SELECT * FROM \\"users\\" WHERE (\\"id\\" = :user_id);",
}
`;
